lib/vsc/administration/vo.py

```diff
@@ -220,76 +228,88 @@ def _create_fileset(self, filesystem_name, path, parent_fileset=None, fileset_na
         except KeyError:
-            logging.exception("Trying to access non-existent field %s in the data storage dictionary", VSC_DATA)
-        self._create_fileset(fs, path)
+            errmsg = "Trying to access non-existent field %s in the data storage dictionary"
+            logging.exception(errmsg, VSC_DATA)
+        else:
+            self._create_fileset(storage, path)
 
     def create_data_shared_fileset(self):
         """Create a VO directory for sharing data on the HPC data filesystem. Always set the quota."""
         path = self._data_shared_path()
-        msg = "Trying to access non-existent"
         try:
-            fs = self.storage[self.host_institute][VSC_DATA_SHARED].filesystem
-        except AttributeError:
-            logging.exception("%s attribute 'filesystem' in the shared data storage instance", msg)
+            storage = self.storage[self.host_institute][VSC_DATA_SHARED]
         except KeyError:
-            logging.exception("%s field %s in the shared data storage dictionary", msg, VSC_DATA_SHARED)
-        self._create_fileset(fs, path,
-                             fileset_name=self.sharing_group.vsc_id,
-                             group_owner_id=self.sharing_group.vsc_id_number)
+            errmsg = "Trying to access non-existent field %s in the shared data storage dictionary"
+            logging.exception(errmsg, VSC_DATA_SHARED)
+        else:
+            self._create_fileset(
+                storage,
+                path,
+                fileset_name=self.sharing_group.vsc_id,
+                group_owner_id=self.sharing_group.vsc_id_number,
+            )
 
     def create_scratch_fileset(self, storage_name):
         """Create the VO's directory on the HPC data filesystem. Always set the quota."""
-        msg = "Trying to access non-existent"
+        path = self._scratch_path(storage_name)
         try:
-            path = self._scratch_path(storage_name)
-            if self.storage[self.host_institute][storage_name].version >= (3, 5, 0, 0):
-                self._create_fileset(self.storage[self.host_institute][storage_name].filesystem, path)
-            else:
-                self._create_fileset(self.storage[self.host_institute][storage_name].filesystem, path, 'root')
-        except AttributeError:
-            logging.exception("%s attribute 'filesystem' in the scratch storage instance", msg)
+            storage = self.storage[self.host_institute][storage_name]
         except KeyError:
-            logging.exception("%s field %s in the scratch storage dictionary", msg, storage_name)
+            errmsg = "Trying to access non-existent field %s in the scratch storage dictionary"
+            logging.exception(errmsg, storage_name)
 
-    def _create_vo_dir(self, path):
-        """Create a user owned directory on the GPFS."""
-        self.gpfs.make_dir(path)
+        if storage.backend == 'gpfs':
+            if storage.version >= (3, 5, 0, 0):
```

Comment:
you can remove this check, there's nothing that old aroudn anynmore

Reply:
done in https://github.com/hpcugent/vsc-administration/pull/134/commits/ebfb2bdb8d26b21d4ee6b782c5bdfe3b35bb9016